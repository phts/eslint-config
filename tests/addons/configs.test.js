'use strict'

const helpers = require('../helpers')

describe('addons/configs', () => {
  helpers.itGeneratesCorrectConfig({
    extends: ['index', 'addons/configs'],
    files: [
      'config/target.js',
      'config/dir/target.js',

      '.target.js',

      'webpack.dev.js',
      'dir/webpack.prod.js',

      'target.config.js',
      'dir/target.config.js',

      'target.conf.js',
      'dir/target.conf.js',
    ],
    for: 'config',
  })

  helpers.itGeneratesCorrectConfig({
    extends: ['index', 'addons/configs'],
    files: ['config/jest.js', 'config/dir/jest.js'],
    for: 'jest setupFilesAfterEnv',
  })
})
