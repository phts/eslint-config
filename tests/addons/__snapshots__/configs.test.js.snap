// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addons/configs generates correct config for config files 1`] = `
{
  "env": {
    "es6": true,
    "node": true,
  },
  "globals": {},
  "ignorePatterns": [
    "playground/",
  ],
  "noInlineConfig": undefined,
  "parser": null,
  "parserOptions": {
    "ecmaVersion": 2018,
    "sourceType": "script",
  },
  "plugins": [
    "eslint-comments",
    "import",
  ],
  "reportUnusedDisableDirectives": undefined,
  "rules": {
    "accessor-pairs": [
      "error",
    ],
    "array-bracket-spacing": [
      "error",
    ],
    "brace-style": [
      "error",
    ],
    "camelcase": [
      "error",
    ],
    "comma-dangle": [
      "error",
      "always-multiline",
    ],
    "comma-spacing": [
      "error",
    ],
    "comma-style": [
      "error",
    ],
    "constructor-super": [
      "error",
    ],
    "curly": [
      "error",
    ],
    "dot-location": [
      "error",
      "property",
    ],
    "eol-last": [
      "error",
    ],
    "eqeqeq": [
      "error",
    ],
    "eslint-comments/disable-enable-pair": [
      "error",
      {
        "allowWholeFile": true,
      },
    ],
    "eslint-comments/no-aggregating-enable": [
      "error",
    ],
    "eslint-comments/no-duplicate-disable": [
      "error",
    ],
    "eslint-comments/no-unlimited-disable": [
      "error",
    ],
    "eslint-comments/no-unused-disable": [
      "error",
    ],
    "eslint-comments/no-unused-enable": [
      "error",
    ],
    "for-direction": [
      "error",
    ],
    "func-call-spacing": [
      "error",
    ],
    "getter-return": [
      "error",
    ],
    "import/export": [
      "error",
    ],
    "import/first": [
      "error",
    ],
    "import/named": [
      "error",
    ],
    "import/newline-after-import": [
      "error",
    ],
    "import/no-absolute-path": [
      "error",
    ],
    "import/no-cycle": [
      "error",
    ],
    "import/no-named-as-default": [
      "error",
    ],
    "import/no-named-as-default-member": [
      "error",
    ],
    "import/no-self-import": [
      "error",
    ],
    "import/no-useless-path-segments": [
      "error",
    ],
    "import/order": [
      "error",
    ],
    "indent": [
      "error",
      2,
    ],
    "jsx-quotes": [
      "error",
      "prefer-double",
    ],
    "key-spacing": [
      "error",
      {
        "afterColon": true,
        "beforeColon": false,
        "mode": "minimum",
      },
    ],
    "keyword-spacing": [
      "error",
    ],
    "max-len": [
      "error",
      {
        "code": 100,
      },
    ],
    "new-cap": [
      "error",
    ],
    "new-parens": [
      "error",
    ],
    "no-array-constructor": [
      "error",
    ],
    "no-async-promise-executor": [
      "error",
    ],
    "no-caller": [
      "error",
    ],
    "no-case-declarations": [
      "error",
    ],
    "no-class-assign": [
      "error",
    ],
    "no-compare-neg-zero": [
      "error",
    ],
    "no-cond-assign": [
      "error",
    ],
    "no-console": [
      "error",
      {
        "allow": [
          "info",
          "warn",
          "error",
        ],
      },
    ],
    "no-const-assign": [
      "error",
    ],
    "no-constant-condition": [
      "error",
    ],
    "no-control-regex": [
      "error",
    ],
    "no-debugger": [
      "error",
    ],
    "no-delete-var": [
      "error",
    ],
    "no-dupe-args": [
      "error",
    ],
    "no-dupe-class-members": [
      "error",
    ],
    "no-dupe-else-if": [
      "error",
    ],
    "no-dupe-keys": [
      "error",
    ],
    "no-duplicate-case": [
      "error",
    ],
    "no-duplicate-imports": [
      "error",
    ],
    "no-empty": [
      "error",
    ],
    "no-empty-character-class": [
      "error",
    ],
    "no-empty-pattern": [
      "error",
    ],
    "no-eval": [
      "error",
    ],
    "no-ex-assign": [
      "error",
    ],
    "no-extra-bind": [
      "error",
    ],
    "no-extra-boolean-cast": [
      "error",
    ],
    "no-extra-parens": [
      "error",
      "all",
      {
        "ignoreJSX": "all",
      },
    ],
    "no-extra-semi": [
      "error",
    ],
    "no-fallthrough": [
      "error",
    ],
    "no-floating-decimal": [
      "error",
    ],
    "no-func-assign": [
      "error",
    ],
    "no-global-assign": [
      "error",
    ],
    "no-implied-eval": [
      "error",
    ],
    "no-import-assign": [
      "error",
    ],
    "no-inner-declarations": [
      "error",
    ],
    "no-invalid-regexp": [
      "error",
    ],
    "no-irregular-whitespace": [
      "error",
    ],
    "no-label-var": [
      "error",
    ],
    "no-labels": [
      "error",
    ],
    "no-lone-blocks": [
      "error",
    ],
    "no-loss-of-precision": [
      "error",
    ],
    "no-misleading-character-class": [
      "error",
    ],
    "no-mixed-spaces-and-tabs": [
      "error",
    ],
    "no-multiple-empty-lines": [
      "error",
    ],
    "no-new-object": [
      "error",
    ],
    "no-new-require": [
      "error",
    ],
    "no-new-symbol": [
      "error",
    ],
    "no-nonoctal-decimal-escape": [
      "error",
    ],
    "no-obj-calls": [
      "error",
    ],
    "no-octal": [
      "error",
    ],
    "no-path-concat": [
      "error",
    ],
    "no-proto": [
      "error",
    ],
    "no-prototype-builtins": [
      "error",
    ],
    "no-redeclare": [
      "error",
    ],
    "no-regex-spaces": [
      "error",
    ],
    "no-return-assign": [
      "error",
    ],
    "no-return-await": [
      "error",
    ],
    "no-self-assign": [
      "error",
    ],
    "no-self-compare": [
      "error",
    ],
    "no-sequences": [
      "error",
    ],
    "no-setter-return": [
      "error",
    ],
    "no-shadow-restricted-names": [
      "error",
    ],
    "no-sparse-arrays": [
      "error",
    ],
    "no-tabs": [
      "error",
    ],
    "no-this-before-super": [
      "error",
    ],
    "no-throw-literal": [
      "error",
    ],
    "no-trailing-spaces": [
      "error",
    ],
    "no-undef": [
      "error",
    ],
    "no-undef-init": [
      "error",
    ],
    "no-unexpected-multiline": [
      "error",
    ],
    "no-unmodified-loop-condition": [
      "error",
    ],
    "no-unneeded-ternary": [
      "error",
    ],
    "no-unreachable": [
      "error",
    ],
    "no-unsafe-finally": [
      "error",
    ],
    "no-unsafe-negation": [
      "error",
    ],
    "no-unsafe-optional-chaining": [
      "error",
    ],
    "no-unused-labels": [
      "error",
    ],
    "no-unused-vars": [
      "error",
    ],
    "no-use-before-define": [
      "error",
    ],
    "no-useless-backreference": [
      "error",
    ],
    "no-useless-call": [
      "error",
    ],
    "no-useless-catch": [
      "error",
    ],
    "no-useless-computed-key": [
      "error",
    ],
    "no-useless-constructor": [
      "error",
    ],
    "no-useless-escape": [
      "error",
    ],
    "no-useless-rename": [
      "error",
    ],
    "no-var": [
      "error",
    ],
    "no-whitespace-before-property": [
      "error",
    ],
    "no-with": [
      "error",
    ],
    "object-curly-spacing": [
      "error",
      "never",
    ],
    "object-property-newline": [
      "error",
      {
        "allowAllPropertiesOnSameLine": true,
        "allowMultiplePropertiesPerLine": false,
      },
    ],
    "one-var": [
      "error",
      "never",
    ],
    "one-var-declaration-per-line": [
      "error",
    ],
    "padded-blocks": [
      "error",
      "never",
    ],
    "prefer-const": [
      "error",
    ],
    "quote-props": [
      "error",
      "as-needed",
    ],
    "quotes": [
      "error",
      "single",
      {
        "allowTemplateLiterals": false,
        "avoidEscape": true,
      },
    ],
    "require-yield": [
      "error",
    ],
    "rest-spread-spacing": [
      "error",
    ],
    "semi": [
      "error",
      "never",
    ],
    "semi-spacing": [
      "error",
    ],
    "space-before-blocks": [
      "error",
    ],
    "space-before-function-paren": [
      "error",
      {
        "anonymous": "always",
        "asyncArrow": "always",
        "named": "never",
      },
    ],
    "space-in-parens": [
      "error",
    ],
    "space-infix-ops": [
      "error",
    ],
    "space-unary-ops": [
      "error",
    ],
    "spaced-comment": [
      "error",
    ],
    "strict": [
      "error",
    ],
    "template-curly-spacing": [
      "error",
    ],
    "use-isnan": [
      "error",
    ],
    "valid-typeof": [
      "error",
    ],
    "wrap-iife": [
      "error",
    ],
    "yield-star-spacing": [
      "error",
      "before",
    ],
    "yoda": [
      "error",
    ],
  },
  "settings": {},
}
`;

exports[`addons/configs generates correct config for jest setupFilesAfterEnv files 1`] = `
{
  "env": {
    "es6": true,
    "node": true,
  },
  "globals": {},
  "ignorePatterns": [
    "playground/",
  ],
  "noInlineConfig": undefined,
  "parser": null,
  "parserOptions": {
    "ecmaVersion": 2018,
    "sourceType": "module",
  },
  "plugins": [
    "eslint-comments",
    "import",
  ],
  "reportUnusedDisableDirectives": undefined,
  "rules": {
    "accessor-pairs": [
      "error",
    ],
    "array-bracket-spacing": [
      "error",
    ],
    "brace-style": [
      "error",
    ],
    "camelcase": [
      "error",
    ],
    "comma-dangle": [
      "error",
      "always-multiline",
    ],
    "comma-spacing": [
      "error",
    ],
    "comma-style": [
      "error",
    ],
    "constructor-super": [
      "error",
    ],
    "curly": [
      "error",
    ],
    "dot-location": [
      "error",
      "property",
    ],
    "eol-last": [
      "error",
    ],
    "eqeqeq": [
      "error",
    ],
    "eslint-comments/disable-enable-pair": [
      "error",
      {
        "allowWholeFile": true,
      },
    ],
    "eslint-comments/no-aggregating-enable": [
      "error",
    ],
    "eslint-comments/no-duplicate-disable": [
      "error",
    ],
    "eslint-comments/no-unlimited-disable": [
      "error",
    ],
    "eslint-comments/no-unused-disable": [
      "error",
    ],
    "eslint-comments/no-unused-enable": [
      "error",
    ],
    "for-direction": [
      "error",
    ],
    "func-call-spacing": [
      "error",
    ],
    "getter-return": [
      "error",
    ],
    "import/export": [
      "error",
    ],
    "import/first": [
      "error",
    ],
    "import/named": [
      "error",
    ],
    "import/newline-after-import": [
      "error",
    ],
    "import/no-absolute-path": [
      "error",
    ],
    "import/no-cycle": [
      "error",
    ],
    "import/no-named-as-default": [
      "error",
    ],
    "import/no-named-as-default-member": [
      "error",
    ],
    "import/no-self-import": [
      "error",
    ],
    "import/no-useless-path-segments": [
      "error",
    ],
    "import/order": [
      "error",
    ],
    "indent": [
      "error",
      2,
    ],
    "jsx-quotes": [
      "error",
      "prefer-double",
    ],
    "key-spacing": [
      "error",
      {
        "afterColon": true,
        "beforeColon": false,
        "mode": "minimum",
      },
    ],
    "keyword-spacing": [
      "error",
    ],
    "max-len": [
      "error",
      {
        "code": 100,
      },
    ],
    "new-cap": [
      "error",
    ],
    "new-parens": [
      "error",
    ],
    "no-array-constructor": [
      "error",
    ],
    "no-async-promise-executor": [
      "error",
    ],
    "no-caller": [
      "error",
    ],
    "no-case-declarations": [
      "error",
    ],
    "no-class-assign": [
      "error",
    ],
    "no-compare-neg-zero": [
      "error",
    ],
    "no-cond-assign": [
      "error",
    ],
    "no-console": [
      "error",
      {
        "allow": [
          "info",
          "warn",
          "error",
        ],
      },
    ],
    "no-const-assign": [
      "error",
    ],
    "no-constant-condition": [
      "error",
    ],
    "no-control-regex": [
      "error",
    ],
    "no-debugger": [
      "error",
    ],
    "no-delete-var": [
      "error",
    ],
    "no-dupe-args": [
      "error",
    ],
    "no-dupe-class-members": [
      "error",
    ],
    "no-dupe-else-if": [
      "error",
    ],
    "no-dupe-keys": [
      "error",
    ],
    "no-duplicate-case": [
      "error",
    ],
    "no-duplicate-imports": [
      "error",
    ],
    "no-empty": [
      "error",
    ],
    "no-empty-character-class": [
      "error",
    ],
    "no-empty-pattern": [
      "error",
    ],
    "no-eval": [
      "error",
    ],
    "no-ex-assign": [
      "error",
    ],
    "no-extra-bind": [
      "error",
    ],
    "no-extra-boolean-cast": [
      "error",
    ],
    "no-extra-parens": [
      "error",
      "all",
      {
        "ignoreJSX": "all",
      },
    ],
    "no-extra-semi": [
      "error",
    ],
    "no-fallthrough": [
      "error",
    ],
    "no-floating-decimal": [
      "error",
    ],
    "no-func-assign": [
      "error",
    ],
    "no-global-assign": [
      "error",
    ],
    "no-implied-eval": [
      "error",
    ],
    "no-import-assign": [
      "error",
    ],
    "no-inner-declarations": [
      "error",
    ],
    "no-invalid-regexp": [
      "error",
    ],
    "no-irregular-whitespace": [
      "error",
    ],
    "no-label-var": [
      "error",
    ],
    "no-labels": [
      "error",
    ],
    "no-lone-blocks": [
      "error",
    ],
    "no-loss-of-precision": [
      "error",
    ],
    "no-misleading-character-class": [
      "error",
    ],
    "no-mixed-spaces-and-tabs": [
      "error",
    ],
    "no-multiple-empty-lines": [
      "error",
    ],
    "no-new-object": [
      "error",
    ],
    "no-new-require": [
      "error",
    ],
    "no-new-symbol": [
      "error",
    ],
    "no-nonoctal-decimal-escape": [
      "error",
    ],
    "no-obj-calls": [
      "error",
    ],
    "no-octal": [
      "error",
    ],
    "no-path-concat": [
      "error",
    ],
    "no-proto": [
      "error",
    ],
    "no-prototype-builtins": [
      "error",
    ],
    "no-redeclare": [
      "error",
    ],
    "no-regex-spaces": [
      "error",
    ],
    "no-return-assign": [
      "error",
    ],
    "no-return-await": [
      "error",
    ],
    "no-self-assign": [
      "error",
    ],
    "no-self-compare": [
      "error",
    ],
    "no-sequences": [
      "error",
    ],
    "no-setter-return": [
      "error",
    ],
    "no-shadow-restricted-names": [
      "error",
    ],
    "no-sparse-arrays": [
      "error",
    ],
    "no-tabs": [
      "error",
    ],
    "no-this-before-super": [
      "error",
    ],
    "no-throw-literal": [
      "error",
    ],
    "no-trailing-spaces": [
      "error",
    ],
    "no-undef": [
      "error",
    ],
    "no-undef-init": [
      "error",
    ],
    "no-unexpected-multiline": [
      "error",
    ],
    "no-unmodified-loop-condition": [
      "error",
    ],
    "no-unneeded-ternary": [
      "error",
    ],
    "no-unreachable": [
      "error",
    ],
    "no-unsafe-finally": [
      "error",
    ],
    "no-unsafe-negation": [
      "error",
    ],
    "no-unsafe-optional-chaining": [
      "error",
    ],
    "no-unused-labels": [
      "error",
    ],
    "no-unused-vars": [
      "error",
    ],
    "no-use-before-define": [
      "error",
    ],
    "no-useless-backreference": [
      "error",
    ],
    "no-useless-call": [
      "error",
    ],
    "no-useless-catch": [
      "error",
    ],
    "no-useless-computed-key": [
      "error",
    ],
    "no-useless-constructor": [
      "error",
    ],
    "no-useless-escape": [
      "error",
    ],
    "no-useless-rename": [
      "error",
    ],
    "no-var": [
      "error",
    ],
    "no-whitespace-before-property": [
      "error",
    ],
    "no-with": [
      "error",
    ],
    "object-curly-spacing": [
      "error",
      "never",
    ],
    "object-property-newline": [
      "error",
      {
        "allowAllPropertiesOnSameLine": true,
        "allowMultiplePropertiesPerLine": false,
      },
    ],
    "one-var": [
      "error",
      "never",
    ],
    "one-var-declaration-per-line": [
      "error",
    ],
    "padded-blocks": [
      "error",
      "never",
    ],
    "prefer-const": [
      "error",
    ],
    "quote-props": [
      "error",
      "as-needed",
    ],
    "quotes": [
      "error",
      "single",
      {
        "allowTemplateLiterals": false,
        "avoidEscape": true,
      },
    ],
    "require-yield": [
      "error",
    ],
    "rest-spread-spacing": [
      "error",
    ],
    "semi": [
      "error",
      "never",
    ],
    "semi-spacing": [
      "error",
    ],
    "space-before-blocks": [
      "error",
    ],
    "space-before-function-paren": [
      "error",
      {
        "anonymous": "always",
        "asyncArrow": "always",
        "named": "never",
      },
    ],
    "space-in-parens": [
      "error",
    ],
    "space-infix-ops": [
      "error",
    ],
    "space-unary-ops": [
      "error",
    ],
    "spaced-comment": [
      "error",
    ],
    "strict": [
      "error",
    ],
    "template-curly-spacing": [
      "error",
    ],
    "use-isnan": [
      "error",
    ],
    "valid-typeof": [
      "error",
    ],
    "wrap-iife": [
      "error",
    ],
    "yield-star-spacing": [
      "error",
      "before",
    ],
    "yoda": [
      "error",
    ],
  },
  "settings": {},
}
`;
